init.lua

```diff
@@ -21,6 +18,85 @@ local hanggliding_players = {}
 local physics_overrides = {}
 local hud_overlay_ids = {}
 
+local function check_physics_engine()
+    if core.get_modpath("pova") then
+        return "pova"
+    elseif core.get_modpath("player_monoids") then
+        return "monoids"
+    end
+end
+
+local physics = check_physics_engine()
+
+-- Empty functions
+local set_physics_overrides
+local remove_physics_overrides
+local physics_id = "hangglider:glider"
+
+-- Filling functions with the proper physics logic.
+local function make_builtin_overrides()
+    local physics_overrides = {}
+    return {
+        set = function(player, overrides)
+            local name = player:get_player_name()
+            local def = player:get_physics_override()
+            physics_overrides[name] = physics_overrides[name] or {
+                physics = {speed = def.speed, jump = def.jump, gravity = def.gravity},
+                deltas  = {speed = 0, jump = 0, gravity = 0},
+            }
+            local delta = {
+                speed   = (overrides.speed or def.speed) - def.speed,
+                jump    = (overrides.jump or def.jump) - def.jump,
+                gravity = (overrides.gravity or def.gravity) - def.gravity,
+            }
+            for k,v in pairs(delta) do
+                physics_overrides[name].deltas[k] = physics_overrides[name].deltas[k] + v
+            end
+            player:set_physics_override({
+                speed   = def.speed   + delta.speed,
+                jump    = def.jump    + delta.jump,
+                gravity = def.gravity + delta.gravity,
+            })
+        end,
+        remove = function(player)
+            local name = player:get_player_name()
+            local def = player:get_physics_override()
+            local ovr = physics_overrides[name]
+            if ovr then
+                player:set_physics_override({
+                    speed   = def.speed   - ovr.deltas.speed,
+                    jump    = def.jump    - ovr.deltas.jump,
+                    gravity = def.gravity - ovr.deltas.gravity,
+                })
+                physics_overrides[name] = nil
+            else
+                player:set_physics_override({speed=1, jump=1, gravity=1})
+            end
+        end
+    }
+end
+
+local overrides = (physics == "pova" and {
+    set = function(player, o)
+        local name = player:get_player_name()
+        pova.add_override(name, physics_id, {jump=o.jump or 0, speed=o.speed, gravity=o.gravity})
+        pova.do_override(player)
+    end,
+    remove = function(player)
+        pova.del_override(player:get_player_name(), physics_id)
+        pova.do_override(player)
+    end
+}) or (physics == "monoids" and {
+    set = function(player, o)
+        for k,v in pairs(o) do player_monoids[k]:add_change(player, v, physics_id) end
+    end,
+    remove = function(player)
+        for _,k in ipairs({"jump","speed","gravity"}) do player_monoids[k]:del_change(player, physics_id) end
```

Comment:
``for _,k in ipairs({"jump","speed","gravity"}) do player_monoids[k]:del_change(player, physics_id) end`` -> ``for _, k in ipairs({ "jump", "speed", "gravity" }) do player_monoids[k]:del_change(player, physics_id) end``